Android: TableViewRowProxy.getIndexInSection()' on a null object

### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

Error:

```
[ERROR] TiExceptionHandler: (main) [19183,19239] Attempt to invoke virtual method 'int ti.modules.titanium.ui.TableViewRowProxy.getIndexInSection()' on a null object reference
[ERROR] TiExceptionHandler:
[ERROR] TiExceptionHandler:     ti.modules.titanium.ui.widget.tableview.TiTableView.generateScrollPayload(TiTableView.java:345)
[ERROR] TiExceptionHandler:     ti.modules.titanium.ui.widget.tableview.TiTableView$2.onScrollStateChanged(TiTableView.java:102)
[ERROR] TiExceptionHandler:     androidx.recyclerview.widget.RecyclerView.dispatchOnScrollStateChanged(RecyclerView.java:5384)
[ERROR] TiExceptionHandler:     androidx.recyclerview.widget.RecyclerView.setScrollState(RecyclerView.java:1618)
[ERROR] TiExceptionHandler:     androidx.recyclerview.widget.RecyclerView.stopScroll(RecyclerView.java:2626)
[ERROR] TiExceptionHandler:     ti.modules.titanium.ui.widget.listview.TiNestedRecyclerView$1.onInterceptTouchEvent(TiNestedRecyclerView.java:55)
```

### Steps to reproduce

```js
const win = Ti.UI.createWindow();
const tableView = Ti.UI.createTableView({});
var tableData = [];

setInterval(function() {
	for (var i = 1; i <= 10; i++) {
		var entryRow = Ti.UI.createTableViewRow({
			height: 40
		});
		var entryNameLabel = Ti.UI.createLabel({
			text: "test"
		});
		entryRow.add(entryNameLabel);
		tableData.push(entryRow);
	}
	tableView.data = tableData;
}, 1000);

win.add(tableView);
win.open();
```
let it running for a while and always scroll up and down.

### Platform

Android

### SDK version you are using

12.0.0

### Alloy version you are using

_No response_

--- a/android/modules/ui/src/java/ti/modules/titanium/ui/widget/tableview/TiTableView.java
+++ b/android/modules/ui/src/java/ti/modules/titanium/ui/widget/tableview/TiTableView.java
@@ -342,7 +342,10 @@ public class TiTableView extends TiSwipeRefreshLayout implements OnSearchChangeL
 			final TableViewHolder firstVisibleHolder =
 				(TableViewHolder) recyclerView.getChildViewHolder(firstVisibleView);
 			final TableViewRowProxy firstVisibleProxy = (TableViewRowProxy) firstVisibleHolder.getProxy();
-			final int firstVisibleIndex = firstVisibleProxy.getIndexInSection();
+			int firstVisibleIndex = -1;
+			if (firstVisibleProxy != null) {
+				firstVisibleIndex = firstVisibleProxy.getIndexInSection();
+			}
 			payload.put(TiC.PROPERTY_FIRST_VISIBLE_ITEM, firstVisibleIndex);
 		}
 
@@ -746,7 +749,12 @@ public class TiTableView extends TiSwipeRefreshLayout implements OnSearchChangeL
 		}
 
 		// Notify adapter of changes on UI thread.
-		this.adapter.update(this.rows, force);
+		recyclerView.post(new Runnable() {
+			public void run()
+			{
+				adapter.update(rows, force);
+			}
+		});
 
 		// FIXME: This is not an ideal workaround for an issue where recycled items that were in focus
 		//        lose their focus when the data set changes. There are improvements to be made here.

